wdl-engine/src/value.rs

```diff
@@ -1183,7 +1184,12 @@ impl PrimitiveValue {
         }
 
         // Perform the join
-        if let Ok(s) = to.join(path.as_str()).into_os_string().into_string() {
+        if let Ok(s) = to
+            .join(path.as_str())
+            .clean()
```

Comment:
Here is where the missing call to `clean` was.

Sorry for the extraneous noise around preferring the `clean` from the extension trait.